generative/maisi/scripts/find_masks.py

```diff
@@ -0,0 +1,155 @@
+        "chest": 1,
+        "thorax": 1,
+        "chest/thorax": 1,
+        "abdomen": 2,
+        "pelvis": 3,
+        "lower": 3,
+        "pelvis/lower": 3,
+    }
+
+    # perform mapping
+    body_region_indices = []
+    for region in body_region:
+        normalized_region = region.lower()  # norm str to lower case
+        if normalized_region not in region_mapping_maisi:
+            raise ValueError(f"Invalid region: {normalized_region}")
+        body_region_indices.append(region_mapping_maisi[normalized_region])
+
+    return body_region_indices
+
+
+def find_masks(
+    body_region: str | Sequence[str],
+    anatomy_list: int | Sequence[int],
+    spacing: Sequence[float] | float = 1.0,
+    output_size: Sequence[int] = [512, 512, 512],
+    check_spacing_and_output_size: bool = False,
+    database_filepath: str = "./data/database.json",
+    mask_foldername: str = "./data/masks/",
+):
+    """
+    Find candidate masks that fullfills all the requirements.
+    They shoud contain all the body region in `body_region`, all the anatomies in `anatomy_list`.
+    If there is no tumor asked in `anatomy_list`, we also expect the candidate masks to be tumor free.
+    If check_spacing_and_output_size is True, the candidate masks need to have the expected `spacing` and `output_size`.
+    Args:
+        body_region: list of input body region string. If single str, will be converted to list of str.
+                     The found candidate mask will include these body regions.
+        anatomy_list: list of input anatomy. The found candidate mask will include these anatomies.
+        spacing: list of three floats, voxel spacing. If providing a single number, will use it for all the three dimensions.
+        output_size: list of three int, expected candidate mask spatial size.
+        check_spacing_and_output_size: whether we expect candidate mask to have spatial size of `output_size` and voxel size of `spacing`.
+        database_filepath: path for the json file that stores the information of all the candidate masks.
+        mask_foldername: directory that saves all the candidate masks.
+    Return:
+        candidate_masks, list of dict, each dict contains information of one candidate mask that fullfills all the requirements.
+    """
+    # check and preprocess input
+    body_region = convert_body_region(body_region)
+
+    if isinstance(anatomy_list, int):
+        anatomy_list = [anatomy_list]
+
+    spacing = ensure_tuple_rep(spacing, 3)
+
+    if not os.path.exists(mask_foldername):
+        zip_file_path = mask_foldername + ".zip"
+
+        if not os.path.isfile(zip_file_path):
+            raise ValueError(f"Please download {zip_file_path} following the instruction in ./data/README.md.")
+
+        with zipfile.ZipFile(zip_file_path, "r") as zip_ref:
+            print(f"Extracting {zip_file_path}...")
+            zip_ref.extractall(path=mask_foldername)
+        print(f"Unzipped {zip_file_path} to {mask_foldername}.")
```

Comment:
Looks we can use `extractall` here. 
https://github.com/Project-MONAI/MONAI/blob/58106a6255f5936d26e479d11d3a34e3d536c112/monai/apps/utils.py#L245

BTW, if `zip_file_path` is a file which can be directly download, you can also consider using:
https://github.com/Project-MONAI/MONAI/blob/58106a6255f5936d26e479d11d3a34e3d536c112/monai/apps/utils.py#L304

Reply:
Thank you! Done. We may prefer to ask user to download it manually by run `python -m monai.bundle download_large_files --bundle_path .` . What do you think?

Reply:
Sure, that also works. Thanks.